### ISSUE TITLE
Autocorrelation 

### ISSUE THREAD

**Issue by laviniarossimori** [zhouhaoyi/Informer2020#182]
Hi! I want to predict a government bond with a very high autocorrelation ( ACF close to one until a lag of 50). When I use your model I found a very similar pattern for trues and preds in the test phase but I think I shouldn't find it since it is impossible to predict using only the past value (it should be like a random walk) . It is a problem given by autocorrelation? Do you please have any advice to transform my target or another solution or suggestion? I try to use the difference from one day to another as the target but in this case, there are few high peaks that the model doesn't predict. 

Thank you!

**Comment by zhouhaoyi** [comment]
Hi!
Autocorrelation should not be the problem. You can upload a sample of your data and we may check it out.
Moreover, if your task is highly autocorrelation. You should use the traditional decoder rather than the generative decoder. But we found the generative decoder performs quite well in our experiments.

**Comment by laviniarossimori** [comment]
Hi! Thank you for your suggestion and your answer!
I can't share the real data but I found similar data even if with a shorter history. 
[btp.csv](https://github.com/zhouhaoyi/Informer2020/files/6770600/btp.csv)

In the plot preds and trues I obtain this kind of plot that I attach and I think an index market couldn't be predicted as well without another features. 
![Schermata da 2021-07-06 16-11-01](https://user-images.githubusercontent.com/83398203/124615152-467bf200-de75-11eb-8bb7-ffe1f4b778b1.png)

**Comment by zhouhaoyi** [comment]
Thanks for your example. I think this may be caused by the period pattern of the input sequences. I am confused with the term index market, could you please give me more explanation on that?

**Comment by gouzi1209** [comment]
@laviniarossimori @zhouhaoyi Hello, can you ask a question. I am also currently working on the problem of univariate time prediction. The text structure of the data is similar to the file (btp.csv) you sent earlier. Which parameters in the code have you modified to enable univariate prediction? I modified args.features=='S', and used "Dataset_Custom", but there are still some problems, mse=nan in my test set. Looking forward to your advice and reply.

**Comment by jkarolczak** [comment]
Hi @gouzi1209 I'm not the author but your description suggests, that you have some `None`s in your dataset. Check if there is any line with a missing date or value in your file. If so, drop such lines. It should fix your issue.
Parameters like enc_in, dec_in also may differ, as you put a single value, not 7 of them.

**Comment by jkarolczak** [comment]
@laviniarossimori did you predict on data shown to the model (iow training set)? If so it's highly possible to perfectly fit the data, as it was known and you gave sufficiently much time for training. It's not only the case of this particular model, but a general observation that may be made for many, probably even most, forecasting models. 
I hope I'm not wrong, if so, dear all, please, correct me.

**Comment by laviniarossimori** [comment]
> I am confused with the term index market, could you please give me more explanation on that?

Yes of course. the target is the spread, it is the difference between the italian bond yield (BTP) and the german one(BUND). It is a way to understend how italian economy is going as compered to the german one is always  stable. The spread can change at every hour for a lot of different reason, so we could never had a fixed period of time in the timeseries.
